### PR TITLE
Add new workflows

### LINKED ISSUE #20: Add remaining workflows
Some other things that are nice and not too hard to setup

https://github.com/catalystneuro/neuroconv/blob/main/.github/workflows/codespell.yml

https://github.com/catalystneuro/neuroconv/blob/main/.github/workflows/auto-publish.yml (I'll set a global PYPI tokenn)

and since we have a CHANGELOG: https://github.com/catalystneuro/neuroconv/blob/main/.github/workflows/assess-file-changes.yml

### PR DESCRIPTION
**Pull request by garrettmflynn**
fix #20

### CODE REVIEW

File: .github/workflows/assess-file-changes.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        uses: tj-actions/changed-files@v29.0.4
+
+      - name: Assess Source Code Changes
+        id: assess-changes
+        run: |
+          echo "SOURCE_CHANGED=false" >> $GITHUB_OUTPUT
+          echo "CHANGELOG_UPDATED=false" >> $GITHUB_OUTPUT
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            echo $file
+            if [[ $file == "src/"* || $file == "tests/"* || $file == "requirements-minimal.txt" || $file == "requirements-testing.txt" || $file == "setup.py" || $file == ".github/"* ]]
```
Comment (CodyCBakerPhD) [inline .github/workflows/assess-file-changes.yml]:
```suggestion
            if [[ $file == "src/"* || $file == "tests/"* || $file == ".github/"* ]]
```

Shortening; the pyproject approach really saves us a lot of space

File: .github/workflows/auto-publish.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade build
+          python3 -m pip install --upgrade twine
```
Comment (CodyCBakerPhD) [inline .github/workflows/auto-publish.yml]: Hmm do you know why the third one is `python3` but not the others?

File: .github/workflows/auto-publish.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Build package
+        run: |
+          python3 -m build
```
Comment (CodyCBakerPhD) [inline .github/workflows/auto-publish.yml]:
```suggestion
        run: python3 -m build
```

Single line can be collapsed

File: .github/workflows/codespell.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (CodyCBakerPhD) [inline .github/workflows/codespell.yml]:
```suggestion
```

Hmm musta missed this one, not in any of our other workflow styles

File: .github/workflows/auto-publish.yml (hunk 2 above)
Comment (garrettmflynn) [reply .github/workflows/auto-publish.yml]: Just forgot to remove that when I copy-pasted this from the documentation I was reading.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
LGTM then, as with devops of this nature, if there are bugs after merge we'll just sort them out

**Pull request merged**